### PR TITLE
Twitter zk certificate

### PR DESCRIPTION
**Pull request by anhdungle93**
PR to add twitterZkCertificate circuits
- Generalize zkCertificateRegistry
- Add scripts, circuits, circuit tests for twitterZkCertificate (similar to the ones for zkKYC)

### CODE REVIEW

**Review by zwilling — CHANGES_REQUESTED** [review]
Can you please check the reasons for the pipeline failing? Something in `yarn build`.
Also, have you ran `yarn lint:fix` to fix the formatting?

File: packages/zk-certificates/circuits/twitterZkCertificate.circom [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    // provider's EdDSA signature of the leaf hash
+    signal input providerS;
+    signal input providerR8x;
+    signal input providerR8y;
+
+    // variables related to the merkle proof
+    signal input pathElements[levels];
+    signal input leafIndex;
+    signal input root;
+    signal input currentTime;
```
[the remark this answers is not in the document]
Comment (anhdungle93) [reply packages/zk-certificates/circuits/twitterZkCertificate.circom:44]: empty line added

File: packages/zk-certificates/circuits/twitterZkCertificate.circom [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    // EdDSA signature of the user address
+    signal input s2;
+    signal input r8x2;
+    signal input r8y2;
+
+    // pub key of the provider
+    signal input providerAx;
+    signal input providerAy;
+
+    signal output userPubKey[2]; // becomes public as part of the output to check that it corresponds to user address
```
[the remark this answers is not in the document]
Comment (anhdungle93) [reply packages/zk-certificates/circuits/twitterZkCertificate.circom:67]: I also thought so, but wasn't sure.

File: packages/zk-certificates/contracts/GuardianRegistry.sol [hunk 3]
```diff
@@ -28,7 +32,9 @@ contract GuardianRegistry is Ownable {
         _;
     }
 
-    constructor() Ownable(msg.sender) {}
+    constructor(string memory _description) Ownable(msg.sender) {
```
[the remark this answers is not in the document]
Comment (anhdungle93) [reply packages/zk-certificates/contracts/GuardianRegistry.sol:35]: changed that and in some other scripts as well

File: packages/galactica-types/src/zkCertStandard.ts [hunk 4]
```diff
@@ -83,3 +84,35 @@ export const humanIDFieldOrder = [
...
+export type TwitterZkCertificateContent = {
+  accountId: string;
+  creationTime: string;
+  location: string;
+  verified: string;
+  followersCount: string;
+  friendsCount: string;
+  likesCount: string;
+  postsCount: string;
+  expirationTime: string;
```
[the remark this answers is not in the document]
Comment (anhdungle93) [reply packages/galactica-types/src/zkCertStandard.ts:101]: I think we can, if zkCertificate is meant to last forever we can put it as the max int of circom.

[the remark this answers is not in the document]
Comment (zwilling) [reply packages/galactica-types/src/zkCertStandard.ts:100]: Changed it in the other branch.

**Pull request merged**